lib/dx/error/timeout.ex

```diff
@@ -0,0 +1,7 @@
+defmodule Dx.Error.Timeout do
+  defexception []
+
+  def message(_error) do
+    "Timeout occurred"
```

Comment:
I think we can safely assume that the timeout occurred in `dataloader`. Or, if you think it should be double-checked, you can return something more specific from the loader module ([here](https://github.com/dx-beam/dx/blob/v0.3.0/lib/dx/loaders/dataloader.ex#L15-L27)).

Then, we can give the user a bit more context in the error:

```suggestion
    "A timeout occurred while loading the data required."
```

When #21 lands, this _could_ (optionally) even mention what timeout threshold applied and how to change it.